infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -95,8 +92,10 @@
      */
     public ManagementServer(ServerContext serverContext) {
 
-        this.opts = serverContext.getServerConfig();
-        this.localEndpoint = this.opts.get("--address") + ":" + this.opts.get("<port>");
+        this.managementServerConfig = ManagementServerConfig.parse(serverContext.getServerConfig());
+
+        this.localEndpoint = managementServerConfig.getAddress() + ":" +
```

Comment:
It's much better to use NodeLocator instead of host+port, would be great if you replace host+port to NodeLocator here and in ManagementServerConfig

Reply:
At this point I don't want to change the fields of ManagementServer so I'll keep the endpoint as is. The ManagementServerConfiguration however is changed to be based on NodeLocator.